src/interface/Interface.jl

```diff
@@ -233,9 +335,69 @@ function transitions_init(
     N::Integer;
     kwargs...
 ) where {ModelType<:Sampleable, SamplerType<:AbstractSampler}
-    @warn "No transitions_init function has been implemented
-           for objects of types $(typeof(ℓ)) and $(typeof(s))"
-    return Vector(undef, N)
+    return Vector{transition_type(s)}(undef, N)
+end
+
+"""
+    callback(
+        rng::AbstractRNG,
+        ℓ::ModelType,
+        s::SamplerType,
+        N::Integer,
+        iteration::Integer,
+        cb::CallbackType;
+        kwargs...
+    )
+
+`callback` is called after every sample run, and allows you to run some function on a 
+subtype of `AbstractCallback`. Typically this is used to increment a progress meter, show a 
+plot of parameter draws, or otherwise provide information about the sampling process to the user.
+
+By default, `ProgressMeter` is used to show the number of samples remaning.
+"""
+function callback(
+    rng::AbstractRNG,
+    ℓ::ModelType,
+    s::SamplerType,
+    N::Integer,
+    iteration::Integer,
+    t::TransitionType,
+    cb::CallbackType;
+    progress::Bool=true,
+    kwargs...
+) where {
+    ModelType<:Sampleable,
+    SamplerType<:AbstractSampler,
+    CallbackType<:AbstractCallback,
+    TransitionType<:AbstractTransition
+}
+    # Default callback behavior.
+    progress && ProgressMeter.next!(cb.p)
+end
+
+function callback(
+    rng::AbstractRNG,
+    ℓ::ModelType,
+    s::SamplerType,
+    N::Integer,
+    iteration::Integer,
+    t::TransitionType,
+    cb::NoCallback;
+    kwargs...
+) where {
+    ModelType<:Sampleable,
+    SamplerType<:AbstractSampler,
+    TransitionType<:AbstractTransition
+}
+    # Do nothing.
 end
 
+"""
+    transition_type(s::SamplerType)
+
+Return the type of `AbstractTransition` that is to be returned by an 
+`AbstractSampler`.
+"""
+transition_type(s::SamplerType) where {SamplerType<:AbstractSampler} = AbstractTransition
```

Comment:
Style: could be re-written as
```julia
transition_type(s::AbstractSampler) = AbstractTransition
```
Docstring should also be updated to reflect the type constraint.